### PR TITLE
[ISSUE #11957] AuthModule add admin exist

### PR DESCRIPTION
**Pull request by godhth**
Please do not create a Pull Request without creating an issue first.

## What is the purpose of the change

https://github.com/alibaba/nacos/issues/11957

## Brief changelog

XX

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

* [ ] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [ ] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [ ] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

File: auth/src/main/java/com/alibaba/nacos/auth/config/AuthModuleStateBuilder.java [hunk 1]
```diff
@@ -39,13 +39,16 @@ public class AuthModuleStateBuilder implements ModuleStateBuilder {
...
+    public static final String AUTH_ADMIN_EXIST = "auth_admin_exist";
+    
     @Override
     public ModuleState build() {
         ModuleState result = new ModuleState(AUTH_MODULE);
         AuthConfigs authConfigs = ApplicationUtils.getBean(AuthConfigs.class);
         result.newState(AUTH_ENABLED, authConfigs.isAuthEnabled());
         result.newState(LOGIN_PAGE_ENABLED, isLoginPageEnabled(authConfigs));
         result.newState(AUTH_SYSTEM_TYPE, authConfigs.getNacosAuthSystemType());
+        result.newState(AUTH_ADMIN_EXIST, authConfigs.isHasGlobalAdminRole());
```
Comment (KomachiSion) [inline auth/src/main/java/com/alibaba/nacos/auth/config/AuthModuleStateBuilder.java:51]: 建议改一下名字， 比如改成auth_admin_request, 或者auth_need_admin 因为插件实现不一定需要admin

File: auth/src/main/java/com/alibaba/nacos/auth/config/AuthConfigs.java [hunk 2]
```diff
@@ -78,7 +78,7 @@ public class AuthConfigs extends Subscriber<ServerConfigChangeEvent> {
     @Value("${" + Constants.Auth.NACOS_CORE_AUTH_ENABLE_USER_AGENT_AUTH_WHITE + ":false}")
     private boolean enableUserAgentAuthWhite;
     
-    private boolean hasGlobalAdminRole;
+    private boolean hasGlobalAdminRole = false;
```
Comment (KomachiSion) [inline auth/src/main/java/com/alibaba/nacos/auth/config/AuthConfigs.java:81]: 这个默认值是不是不需要制定，本身就是false？

**Review by KomachiSion — CHANGES_REQUESTED** [review]

File: auth/src/main/java/com/alibaba/nacos/auth/AuthService.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+ */
+
+package com.alibaba.nacos.auth;
+
+/**
+ * AuthService.
+ *
+ * @author : huangtianhui
+ */
+public interface AuthService {
```
Comment (KomachiSion) [inline auth/src/main/java/com/alibaba/nacos/auth/AuthService.java:24]: 不需要单独添加interface， 参考 isLoginPageEnabled，把方法定义在插件中，在interface中提供default实现（false）即可

**Review by KomachiSion — CHANGES_REQUESTED** [review]

File: auth/src/main/java/com/alibaba/nacos/auth/config/AuthModuleStateBuilder.java [hunk 4]
```diff
@@ -39,16 +41,26 @@ public class AuthModuleStateBuilder implements ModuleStateBuilder {
...
         result.newState(AUTH_SYSTEM_TYPE, authConfigs.getNacosAuthSystemType());
+        AtomicBoolean adminRequest = new AtomicBoolean(true);
+        ApplicationUtils.getBeanIfExist(AuthService.class, authService -> adminRequest.set(authService.isAdminRequest()));
+        result.newState(AUTH_ADMIN_REQUEST, adminRequest.get());
         return result;
     }
     
+    @Override
+    public boolean isCacheable() {
+        return false;
```
Comment (KomachiSion) [inline auth/src/main/java/com/alibaba/nacos/auth/config/AuthModuleStateBuilder.java:61]: 这里最好是可变的， 如果adminRequest是true的时候，就不能缓存，如果是已经是false，那说明：
1. 插件不需要admin
2. admin已存在
此时就不需要再重新生成了，可以缓存了

**Review by KomachiSion — CHANGES_REQUESTED** [review]

File: auth/src/main/java/com/alibaba/nacos/auth/AuthService.java (hunk 3 above)
Comment (godhth) [reply auth/src/main/java/com/alibaba/nacos/auth/AuthService.java]: done

File: auth/src/main/java/com/alibaba/nacos/auth/config/AuthModuleStateBuilder.java (hunk 4 above)
Comment (godhth) [reply auth/src/main/java/com/alibaba/nacos/auth/config/AuthModuleStateBuilder.java]: done

**Review by KomachiSion — APPROVED** [review]

**Comment by KomachiSion** [comment]
There is some conflict with test file.

**Review by KomachiSion — APPROVED** [review]

**Pull request merged**